### PR TITLE
Fix `__str__` of parameterized class

### PR DESCRIPTION
**Pull request by toslunar**
cf. #6287

### CODE REVIEW

File: chainer/testing/parameterized.py [hunk 1]
```diff
@@ -48,8 +48,11 @@ def _parameterize_test_case_generator(base, params):
         cls_name = _make_class_name(base.__name__, i, param)
 
         def __str__(self):
+            # Bind to a new variable.
+            param2 = param
```
Comment (hvy) [inline chainer/testing/parameterized.py]: Do you have to nest this further e.g. in another function that constructs and return this `__str__` object?

Comment (toslunar) [reply chainer/testing/parameterized.py]: You are right.  I'll fix the code.

**Comment by hvy** [comment]
LGTM!

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
Failure is unrelated (see https://github.com/chainer/chainer/issues/8145), LGTM!

**Pull request merged**